wire-schema/src/test/java/com/squareup/wire/schema/SchemaTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import okio.Buffer;
+import okio.Source;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+public final class SchemaTest {
+  @Test public void linkService() throws Exception {
+    Schema schema = new SchemaBuilder()
+        .add("service.proto", ""
+            + "import \"request.proto\";\n"
+            + "import \"response.proto\";\n"
+            + "service Service {\n"
+            + "  rpc Call (Request) returns (Response);\n"
+            + "}\n")
+        .add("request.proto", ""
+            + "message Request {\n"
+            + "}\n")
+        .add("response.proto", ""
+            + "message Response {\n"
+            + "}\n")
+        .build();
+
+    Service service = schema.getService("Service");
+    Rpc call = service.rpc("Call");
+    assertThat(call.requestType()).isEqualTo(schema.getType("Request").name());
+    assertThat(call.responseType()).isEqualTo(schema.getType("Response").name());
+  }
+
+  @Test public void linkMessage() throws Exception {
+    Schema schema = new SchemaBuilder()
+        .add("message.proto", ""
+            + "import \"foo.proto\";\n"
+            + "message Message {\n"
+            + "  optional foo_package.Foo field = 1;\n"
+            + "}\n")
+        .add("foo.proto", ""
+            + "package foo_package;\n"
+            + "message Foo {\n"
+            + "}\n")
+        .build();
+
+    MessageType message = (MessageType) schema.getType("Message");
+    Field field = message.field("field");
+    assertThat(field.type()).isEqualTo(schema.getType("foo_package.Foo").name());
+  }
+
+  @Test public void messageFieldTypeUnknown() throws Exception {
+    try {
+      new SchemaBuilder()
+          .add("message.proto", ""
+              + "message Message {\n"
+              + "  optional foo_package.Foo unknown = 1;\n"
+              + "}\n")
+          .build();
+      fail();
+    } catch (SchemaException expected) {
+      assertThat(expected).hasMessage("unable to resolve foo_package.Foo\n"
```

Comment:
I think they're fine. If I had one change it would be to change the subsequent "for"s to " in" and to indent those lines two spaces.
```
unable to resolve foo_package.Foo
  for field unknown (message.proto at 2:3)
  in message Message (message.proto at 1:1)
```